README.md

```diff
@@ -21,11 +21,22 @@ For example, an important contract not only requires enough signatures, but also
 
 HTSS  has been developed by [Tassa](https://www.openu.ac.il/lists/mediaserver_documents/personalsites/tamirtassa/hss_conf.pdf) and other researchers many years ago. In our implementation, we setup up this theory on TSS(i.e. just replace Lagrange Interpolation to Birkhoff Interpolation).  Meanwhile, our protocol of sign (i.e. GG18 and CCLST20 ) can support two homomorphic encryptions which are Paillier and CL scheme. 
 
-**This code will be audited soon.**
 
 > One of references of HTSS is [A Hierarchical Threshold Signature](https://medium.com/getamis/a-hierarchical-threshold-signature-830683b87873) or See [Example](#Example).
 
 
+## Audit Report:
+Alice has been audited by [Kudelski Security](https://www.kudelskisecurity.com). The details can be found in [here](REPORT_2020-05-19_FINAL_CORRECTED.pdf).
+
+
+## Warning:
+
+Although Alice has been audited, you  should still be careful to use it. 
+1. Using  end-to-end encryption to transfer message between two parts is necessary. 
```

Comment:
"Using  end-to-end" -> "Using end-to-end"

Reply:
"two parts" -> "two parties"?

Reply:
done